scripts/ci/workflows-issues-manager/index.js

```diff
@@ -0,0 +1,130 @@
+const { ISSUES_MANAGER_TAG } = require("./constants.js");
+const { getRepoWorkflows, getRunsForWorkflow } = require("./workflows.js");
+const { getRepoIssues, closeIssue, createIssue} = require("./issues");
+
+
+const checkConclusions = (conclusions) => {
+  return ({ conclusion }) => {
+    return conclusions.includes(conclusion);
+  };
+};
+
+const filterRuns = (runs, conclusions) => {
+  return runs.filter((run) => conclusions.includes(run.conclusion));
+};
+
+const splitWorkflows = async ({ github, context }, workflows) => {
+  let lastKRuns = [];
+  let unstable = [];
+  let stable = [];
+  let permared = [];
+
+  //TODO: make it parallel
+  for (const workflow of workflows) {
+    const { workflow_runs } = await getRunsForWorkflow({ github, context }, workflow);
+    let filteredRuns = filterRuns(workflow_runs, ["success", "failure", "timed_out"]);
+
+    const output = filteredRuns.map(
+      ({ id, name, conclusion, event, head_branch }) => `${id} | ${name} | ${conclusion} | ${event} |${head_branch}`
+    );
+    console.log("FILTERED WORKFLOW RUNS", output);
+
+    lastKRuns.push({
+      workflow,
+      filteredRuns,
+    });
+  }
+
+  const isSuccessful = checkConclusions(["success"]);
+  const isFailure = checkConclusions(["failure", "timed_out"]);
+
+  //TODO: Handle case when filteredRuns is empty
+
+  lastKRuns = lastKRuns.filter(({ filteredRuns }) => filteredRuns.length > 0);
+
+  unstable = lastKRuns.filter(({ filteredRuns }) => filteredRuns.some(isFailure) && !filteredRuns.every(isFailure));
```

Comment:
Looks like unstable means the there is at least one flake in last K runs. Wondering if this could flood open issues; or keep closing and open issues. I do not have a mind about the preferred strategy but we could rollout this gradually (first monitor a subset of workflows or so) to see if it works well.